app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt

```diff
@@ -45,323 +39,454 @@ import androidx.compose.ui.graphics.Color
-      IngredientView(servingsCount, changeServingsCount, currentRecipe)
-    } else {
-      InstructionView(currentRecipe)
-    }
-  }
+
+  Column(
+      modifier = Modifier.fillMaxSize(),
+      verticalArrangement = Arrangement.spacedBy(SMALL_PADDING.dp)) {
+        if (ingredientsView) {
+          IngredientView(currentRecipe)
+        } else {
+          InstructionView(currentRecipe)
+        }
+      }
 }
 
 /** Display of the ingredients and the ability to change the number of servings */
 @Composable
-private fun IngredientView(
-    servingsCount: Int,
-    changeServingsCount: (Int) -> Unit,
-    currentRecipe: Recipe?
-) {
-  var servingsCountVar = servingsCount
+private fun IngredientView(currentRecipe: Recipe) {
+  var servingsCount by remember { mutableIntStateOf(INITIAL_NUMBER_VALUE_PER_RECIPE) }
   Row(
-      modifier = Modifier.fillMaxWidth(),
+      modifier =
+          Modifier.fillMaxWidth().padding(horizontal = PADDING.dp).testTag("ingredientsView"),
       verticalAlignment = Alignment.CenterVertically,
-      horizontalArrangement = Arrangement.Center) {
+      horizontalArrangement = Arrangement.SpaceBetween) {
+        // Text label on the left
         Text(
-            "Servings",
-            textAlign = TextAlign.Left,
-            color = Color.Black,
-            fontSize = 12.sp,
-            modifier = Modifier.width(56.dp).height(18.dp))
-        Spacer(modifier = Modifier.width(190.dp))
-        Row(
-            modifier =
-                Modifier.background(goldenBronze, shape = RoundedCornerShape(25))
-                    .height(25.dp)
-                    .width(53.dp),
-            horizontalArrangement = Arrangement.Center,
-            verticalAlignment = Alignment.CenterVertically) {
-              // Display of the buttons to change the number of servings
-              Button(
-                  onClick = { if (servingsCountVar > 1) changeServingsCount(--servingsCountVar) },
-                  colors = ButtonColors(goldenBronze, Color.Black, goldenBronze, Color.Black),
-                  modifier = Modifier.size(20.dp).testTag("removeServings"),
-                  contentPadding = PaddingValues()) {
-                    Text("-", fontSize = 12.sp)
-                  }
+            text = stringResource(R.string.servings),
+            textAlign = TextAlign.Start,
+            style = MaterialTheme.typography.titleMedium,
+            color = MaterialTheme.colorScheme.onSecondary,
+            modifier = Modifier.weight(1f))
+
+        // Button row on the right
+        Counter(servingsCount) { newCounter -> servingsCount = newCounter }
+      }
+
+  IngredientsList(currentRecipe, servingsCount)
+}
+
+/**
+ * Display of the counter to change the number of servings
+ *
+ * @param servingsCount: The current number of servings
+ * @param onCounterChange: The function to change the number of servings
+ */
+@Composable
+private fun Counter(servingsCount: Int, onCounterChange: (Int) -> Unit) {
```

Comment:
**Accessibility for Servings Count**:
   - Adding a `contentDescription` to the `Text` displaying `servingsCount` could help with accessibility, so screen readers can announce it more clearly.

   **Suggestion**:
   ```kotlin
   Text(
       text = servingsCount.toString(),
       modifier = Modifier
           .testTag("numberServings")
           .width(OVERVIEW_COUNTER_TEXT_SIZE.dp)
           .semantics { contentDescription = "$servingsCount servings" }
   )
   ```

Reply:
Done !